tachyon/math/polynomials/univariate/radix2_evaluation_domain.h

```diff
@@ -118,10 +118,10 @@ class Radix2EvaluationDomain
     ReverseMatrixIndexBits(mat);
   }
 
-  template <typename Derived>
-  CONSTEXPR_IF_NOT_OPENMP RowMajorMatrix<F> CosetLDEBatch(
+  template <typename Derived, typename Derived2>
+  CONSTEXPR_IF_NOT_OPENMP void CosetLDEBatch(
       Eigen::MatrixBase<Derived>& mat, size_t added_bits, F shift,
```

Comment:
I agreed, but the `mat` is changed internally.

Reply:
I think we can take it as value in this case. Since the `mat` will be garbage anyways, taking reference doesn't make sense. We can move if we don't need the matrix or just give it as value and it will be copied.

Reply:
Correct me if I'm wrong, but from what I've seen it doesn't seem like the matrices that are inputted into `CosetLDEBatch()` are used afterward, so perhaps moving mat (&&) would be better 🤔

Reply:
But then we need to be very careful when calling `CosetLDEBatch()` with this approach.
```
CosetLDEBatch(Eigen::MatrixBase<Derived>& mat,

Reply:
If we use && we can't use const lvalue reference which is for example needed in FRI benchmark. If we just take it as value, we can move when we don't need the matrix afterwards.

```cpp
CosetLDEBatch(Eigen::MatrixBase<Derived> mat,...
// possible
CosetLDEBatch(input_ref,...);
// possible
CosetLDEBatch(std::move(input_mat...),...
```

Reply:
I'm not sure how that's safer than the below code. Could you  explain?

```cpp
void CosetLDEBatch(Eigen::MatrixBase<Derived> mat,...)
```